container/build.py

```diff
@@ -99,32 +108,89 @@
+    except AttributeError as e:
+      print("Error getting attributes from map file.\n", e)
+
+  # Gets the project root (for calling bazel targets)
   project_root = subprocess.check_output(
       shlex.split("git rev-parse --show-toplevel")).strip()
-  package = TYPE_PACKAGE_MAP[type_]
-  target = TYPE_TARGET_MAP[type_]
-  tarball = TYPE_TARBALL_MAP[type_]
-
+  package = type_package_map[type_]
+  target = type_target_map[type_]
+  tarball = None
+  if bucket:
+    tarball = type_tarball_map[type_]
+
+  # Gets the base directory of the bazel-toolchains repo (relative to project_root)
+  # This is for referencing yaml files and mounting the project to gcloud.
+  # Allows the bazel-toolchains repo to be cloned in any subdirectory
+  # of another project and bazel_toolchains_base_dir will store the relative path
+  # from the root of that git project to the root of the bazel-toolchains git project.
+  # Ex. If we are in folder foo and the structure looks like this:
+  # foo/
+  #     subdir/
+  #         bazel-toolchains
+  #     ...
+  # 
+  # Then if we call build.py with the terminal running in foo,
+  # bazel_toolchains_base_dir == "subdir/bazel-toolchains"
+  # (This also allows for renaming of the bazel-toolchains folder as the variable
+  # will store the path regardless of the directory names) 
+  os.chdir(os.path.dirname(__file__))
+  bazel_toolchains_base_dir = os.path.relpath(subprocess.check_output(
+      shlex.split("git rev-parse --show-toplevel")).strip(), project_root)
+  
   # We need to start the build from the root of the project, so that we can
   # mount the full root directory (to use bazel builder properly).
   os.chdir(project_root)
-  # We need to run clean to make sure we don't mount local build outputs
-  subprocess.call(["bazel", "clean"])
+
+  # Ensure all BUILD files under /third_party have the right permission.
+  # This is because in some systems the BUILD files under /third_party
+  # (after git clone) will be with permission 640 and the build will
+  # fail in Container Builder.
+  for dirpath, _, files in os.walk("third_party"):
+    for f in files:
+      full_path = os.path.join(dirpath, f)
+      os.chmod(full_path, 0o644)
 
   if local:
     local_build(type_, package, target)
   else:
-    cloud_build(project_root, project, container, tag, async_, bucket, package, target, tarball)
+    # Gets the yaml relative to the bazel-toolchains root, regardless of what directory it was called from
+    # MUST BE UPDATED IF THE YAML FILE IS MOVED
+    config_file = "{}/container/cloudbuild.yaml".format(bazel_toolchains_base_dir)
+    extra_substitution = ",_BUCKET={},_TARBALL={}".format(bucket, tarball)
+    if not bucket:
+      # Gets the yaml relative to the bazel-toolchains root, regardless of what directory it was called from
+      # MUST BE UPDATED IF THE YAML FILE IS MOVED
+      config_file = "{}/container/cloudbuild_no_bucket.yaml".format(bazel_toolchains_base_dir)
+      extra_substitution = ""
```

Comment:
This part should also stay in cloud_build function. Only move things around if necessary.

Reply:
Moved it back